stand/libsa/in_cksum.c

```diff
@@ -50,20 +50,18 @@ __FBSDID("$FreeBSD$");
  * In particular, it should not be this one.
  */
 int
-in_cksum(p, len)
-	void *p;
-	int len;
+in_cksum(void *p, int len)
 {
 	int sum = 0, oddbyte = 0, v = 0;
-	u_char *cp = p;
+	u_char *cp = (u_char *)p;
 
 	/* we assume < 2^16 bytes being summed */
 	while (len > 0) {
 		if (oddbyte) {
 			sum += v + *cp++;
 			len--;
 		}
-		if (((long)cp & 1) == 0) {
+		if (((intptr_t)cp & 1) == 0) {
```

Comment:
This change is good (arguably it should be uintptr_t, but either are good). It should be a separate commit.